### PR TITLE
Add workflows for publishing runners

### PR DESCRIPTION
**Pull request by sub-mod**
Issue https://github.com/mlperf/mlbox/issues/91

### CODE REVIEW

**Comment by sub-mod** [comment]
The CI will not succeed for this PR.

**Comment by swiftdiaries** [comment]
Just a nit on version numbers, do we want to do `v0.0.x` and graduate to a `v0.1-alpha`?

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -31,3 +29,14 @@ jobs:
       uses: pypa/gh-action-pypi-publish@master
       with:
         password: ${{ secrets.PYPI_PASSWORD }}
+
+  dispatch:
```
Comment (swiftdiaries) [inline .github/workflows/python-publish.yml:33]: More of a question, what does dispatch as an event mean?
When is it triggered?

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -31,3 +29,14 @@ jobs:
...
+    needs: deploy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@v1
+        with:
+          token: ${{ secrets.MLCOMMONS_REPO_ACCESS }}
```
Comment (swiftdiaries) [inline .github/workflows/python-publish.yml:40]: We seem to be using a lot of discrete secrets. Do we want to document that in the `.github/workflows/README.md`?

File: .github/workflows/runner-publish.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        pip install setuptools wheel twine
+    - name: Build distribution
+      run: |
+        cd runners/mlcommons_box_ssh
+        python setup.py sdist bdist_wheel
+        cp -avr dist ../../
+    - name: Publish
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.MLCOMMONS_SSH_PYPI_TOKEN }}
```
Comment (swiftdiaries) [inline .github/workflows/runner-publish.yml:31]: If we're using separate secrets and jobs, does it make sense to have separate workflows instead?

File: runners/mlcommons_box_docker/setup.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def schemas():
 
 setup(
     name="mlcommons_box_docker",
-    version="0.1",
+    version="0.2",
```
Comment (swiftdiaries) [inline runners/mlcommons_box_docker/setup.py:41]: Nit on version numbers/cadence.
`v0.0.x` -> `v0.1-alpha` or `v0.x` -> `v1.0-alpha`?

**Comment by sub-mod** [comment]
it can only be 0.0.7 or v0.1-alpha

File: runners/mlcommons_box_singularity/setup.py [hunk 5]
```diff
@@ -38,7 +38,7 @@ def schemas():
 
 setup(
     name="mlcommons_box_singularity",
-    version="0.1",
+    version="0.2",
```
Comment (swiftdiaries) [inline runners/mlcommons_box_singularity/setup.py:41]: Same as above

File: runners/mlcommons_box_ssh/setup.py [hunk 6]
```diff
@@ -38,7 +38,7 @@ def schemas():
 
 setup(
     name="mlcommons_box_ssh",
-    version="0.1",
+    version="0.2",
```
Comment (swiftdiaries) [inline runners/mlcommons_box_ssh/setup.py:41]: Same as above

**Comment by sub-mod** [comment]
v0.2 -> v0.2.x -> v1.0-alpha
@swiftdiaries

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (sub-mod) [reply .github/workflows/python-publish.yml:33]: mlcommons_box triggers runners build

File: .github/workflows/runner-publish.yml (hunk 3 above)
Comment (sub-mod) [reply .github/workflows/runner-publish.yml:31]: The tokens are for PYPI projects. Separate workflows don't solve it.

File: runners/mlcommons_box_docker/setup.py (hunk 4 above)
Comment (sub-mod) [reply runners/mlcommons_box_docker/setup.py:41]: v0.2 -> v0.2.x -> ... -> v1.0-alpha

File: runners/mlcommons_box_singularity/setup.py (hunk 5 above)
Comment (sub-mod) [reply runners/mlcommons_box_singularity/setup.py:41]: v0.2 -> v0.2.x -> ... -> v1.0-alpha

File: runners/mlcommons_box_ssh/setup.py (hunk 6 above)
Comment (sub-mod) [reply runners/mlcommons_box_ssh/setup.py:41]: v0.2 -> v0.2.x -> ... -> v1.0-alpha

File: .github/workflows/python-publish.yml (hunk 2 above)
Comment (sub-mod) [reply .github/workflows/python-publish.yml:40]: will do as a new PR https://github.com/mlperf/mlbox/issues/94

**Comment by sub-mod** [comment]
@swiftdiaries can I merge this ?

**Pull request merged**